hibernate-core/src/test/java/org/hibernate/orm/test/timezones/JDBCTimeZoneZonedTest.java

```diff
@@ -39,19 +42,24 @@ public class JDBCTimeZoneZonedTest {
 		});
 		scope.inSession( s-> {
 			Zoned z = s.find(Zoned.class, id);
-			ZoneId systemZone = ZoneId.systemDefault();
-			ZoneOffset systemOffset = systemZone.getRules().getOffset( Instant.now() );
-			if ( scope.getSessionFactory().getJdbcServices().getDialect() instanceof SybaseDialect) {
+			final Dialect dialect = scope.getSessionFactory().getJdbcServices().getDialect();
+			if ( dialect instanceof SybaseDialect) {
 				// Sybase with jTDS driver has 1/300th sec precision
 				assertEquals( nowZoned.toInstant().truncatedTo(ChronoUnit.SECONDS), z.zonedDateTime.toInstant().truncatedTo(ChronoUnit.SECONDS));
 				assertEquals( nowOffset.toInstant().truncatedTo(ChronoUnit.SECONDS), z.offsetDateTime.toInstant().truncatedTo(ChronoUnit.SECONDS));
 			}
 			else {
-				assertEquals( nowZoned.withZoneSameInstant(systemZone), z.zonedDateTime );
-				assertEquals( nowOffset.withOffsetSameInstant(systemOffset), z.offsetDateTime );
+				assertEquals(
+						DateTimeUtils.roundToDefaultPrecision( nowZoned.toInstant(), dialect ),
+						DateTimeUtils.roundToDefaultPrecision( z.zonedDateTime.toInstant(), dialect )
+				);
+				assertEquals(
+						DateTimeUtils.roundToDefaultPrecision( nowOffset.toInstant(), dialect ),
+						DateTimeUtils.roundToDefaultPrecision( z.offsetDateTime.toInstant(), dialect )
+				);
 			}
-			assertEquals( systemZone, z.zonedDateTime.getZone() );
-			assertEquals( systemOffset, z.offsetDateTime.getOffset() );
+			assertEquals( ZoneId.systemDefault(), z.zonedDateTime.getZone() );
+			assertEquals( now.getOffset().normalized(), z.offsetDateTime.getOffset().normalized() );
```

Comment:
But this isn't testing quite the same thing. 

    getOffset().normalized()

is not quite the same things as:

    getOffset()